File: MiniGame-Java/src/main/java/rs/lab/mges/game/Scene.java
package rs.lab.mges.game;

import io.github.libsdl4j.api.render.SDL_Renderer;

public abstract class Scene {

    protected RabbitGame game;

    public Scene(RabbitGame game) {
        this.game = game;
    }

    void enter() {
    }

    void leave() {
    }

    abstract void update(float dt);

    abstract void draw(SDL_Renderer renderer, float delta);
}

File: MiniGame-Python/src/game/Scene.py
from enum import Enum

from game import RabbitGame

__all__ = ['Scene', 'GameScene']


class GameScene(Enum):
    TITLE = 0,
    GAMEPLAY = 1


class Scene:

    def __init__(self, game: RabbitGame):
        self._game = game

    # Virtual methods
    def enter(self): pass
    def leave(self): pass

    # Abstract methods
    def update(self, dt): raise NotImplementedError()
    def draw(self, renderer, delta): raise NotImplementedError()
